samples/browser/src/main/java/org/mozilla/samples/browser/CustomTabBrowserFragment.kt

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.browser
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.fragment_browser.view.*
+import mozilla.components.feature.customtabs.CustomTabsToolbarFeature
+import mozilla.components.support.base.feature.BackHandler
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.samples.browser.ext.components
+
+class CustomTabBrowserFragment : BaseBrowserFragment(), BackHandler {
+    private val customTabsToolbarFeature = ViewBoundFeatureWrapper<CustomTabsToolbarFeature>()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        val layout = super.onCreateView(inflater, container, savedInstanceState)
+
+        customTabsToolbarFeature.set(
+            feature = CustomTabsToolbarFeature(
+                components.sessionManager,
+                layout.toolbar,
+                sessionId,
+                components.menuBuilder,
+                closeListener = { activity?.finish() }),
+            owner = this,
+            view = layout)
+
+        return layout
+    }
+
+    override fun onBackPressed(): Boolean =
+        super.onBackPressed() || customTabsToolbarFeature.onBackPressed()
```

Comment:
You probably want to evaluate the feature's `onBackPressed` first before super. Same goes for the `WebAppBroserFragment.kt`.

Reply:
This matches the original behaviour where the custom tabs feature was the last to be evaluated. Should I change it?

Reply:
I think it's correct - just looks confusing. I was confused by it too. In this case `super.onBackPressed()` does not call through to the system (like in an activity), but to `BaseBrowserFragment` which will go through the other features first.

Reply:
I updated this method with a comment to describe what's happening and clear up the ambiguity.

Reply:
Gotcha, I see it now. Thanks for clarifying!